src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_params.py

```diff
@@ -56,11 +57,129 @@ def get_hostname_completion_list(prefix, action, parsed_args, **kwargs):  # pyli
+register_cli_argument('webapp deployment source', 'cd_app_type', arg_group='VSTS CD Provider', help='web application framework you used to develop your app', default='AspNetWap', **enum_choice_list(['AspNetWap', 'AspNetCore', 'NodeJSWithGulp', 'NodeJSWithGrunt']))
+register_cli_argument('webapp deployment source', 'cd_account', arg_group='VSTS CD Provider', help='name of the Team Services account to create/use for continuous delivery')
+register_cli_argument('webapp deployment source', 'cd_account_must_exist', arg_group='VSTS CD Provider', help='specifies that the account must already exist. If not specified, the account will be created if it does not already exist (existing accounts are updated)', action='store_true')
+register_cli_argument('webapp deployment source', 'repository_type', help='repository type', default='git', **enum_choice_list(['git', 'mercurial']))
+register_cli_argument('webapp deployment source', 'git_token', help='git access token required for auto sync')
+# end of new ones
+
+# start of old ones #
 register_cli_argument('appservice web', 'slot', options_list=('--slot', '-s'), help="the name of the slot. Default to the productions slot if not specified")
 register_cli_argument('appservice web', 'name', configured_default='web',
                       arg_type=name_arg_type, completer=get_resource_name_completion_list('Microsoft.Web/sites'), id_part='name',
-                      help="name of the web. You can configure the default using `az configure --defaults web=<name>`")
+                      help="name of the web. You can configure the default using 'az configure --defaults web=<name>'")
 register_cli_argument('appservice web create', 'name', options_list=('--name', '-n'), help='name of the new webapp')
+register_cli_argument('appservice web create', 'deployment_local_git', action='store_true', options_list=('--deployment-local-git', '-l'), help='enable local git')
+register_cli_argument('appservice web create', 'deployment_source_url', options_list=('--deployment-source-url', '-u'), help='Git repository URL to link with manual integration')
+register_cli_argument('appservice web create', 'deployment_source_branch', options_list=('--deployment-source-branch', '-b'), help='the branch to deploy')
+register_cli_argument('appservice web create', 'deployment_container_image_name', options_list=('--deployment-container-image-name', '-i'),
+                      help='Linux only. Container image name from Docker Hub, e.g. publisher/image-name:version')
+register_cli_argument('appservice web create', 'startup_file', help="Linux only. The web's startup file")
+register_cli_argument('appservice web create', 'runtime', options_list=('--runtime', '-r'), help='canonicalized web runtime in the format of Framework|Version. For example, PHP|5.6')  # TODO ADD completer
+register_cli_argument('appservice web list-runtimes', 'linux', action='store_true', help='list runtime stacks for linux based webapps')  # TODO ADD completer
```

Comment:
Can we create `CliArgumentType` objects for the arguments that are duplicated between the old command name and new command names?
That way if we have to change something, we do it in one place.